flash/text/question_answering/data.py

```diff
@@ -0,0 +1,1017 @@
+            test_file: The CSV file containing the testing data.
+            predict_file: The CSV file containing the data to use when predicting.
+            train_transform: The dictionary of transforms to use during training which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            val_transform: The dictionary of transforms to use during validation which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            test_transform: The dictionary of transforms to use during testing which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            predict_transform: The dictionary of transforms to use during predicting which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            data_fetcher: The :class:`~flash.core.data.callback.BaseDataFetcher` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`.
+            preprocess: The :class:`~flash.core.data.data.Preprocess` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`. If ``None``, ``cls.preprocess_cls``
+                will be constructed and used.
+            val_split: The ``val_split`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            batch_size: The ``batch_size`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            num_workers: The ``num_workers`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            sampler: The ``sampler`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            preprocess_kwargs: Additional keyword arguments to use when constructing the preprocess. Will only be used
+                if ``preprocess = None``.
+
+        .. note:: The following keyword arguments can be passed through to the preprocess_kwargs
+
+            - backbone: The HF model to be used for the task.
+            - max_source_length: Max length of the sequence to be considered during tokenization.
+            - max_target_length: Max length of each answer to be produced.
+            - padding: Padding type during tokenization. Defaults to 'max_length'.
+            - question_column_name: The key in the JSON file to recognize the question field. Defaults to"question".
+            - context_column_name: The key in the JSON file to recognize the context field. Defaults to "context".
+            - answer_column_name: The key in the JSON file to recognize the answer field. Defaults to "answer".
+            - doc_stride: The stride amount to be taken when splitting up a long document into chunks.
+
+        Returns:
+            The constructed data module.
+
+        Examples::
+
+            data_module = DataModule.from_csv(
```

Comment:
These can all be `QuestionAnsweringData`
```suggestion
            data_module = QuestionAnsweringData.from_csv(
```